### ISSUE TITLE
Using fetch streams API without recursion

### ISSUE THREAD

**Issue by qacollective** [whatwg/streams#958]
Hello,

I'm using the Streams API and have only encountered tutorials / examples which use recursion.  I'm running into a problem which **may** still have the potential to improve the Streams API standard.

I have found that the depth of my code's recursion depends on a variable factor: the chunk/frame size of data sent from the server.  More pointedly, a small delay between broadcasts to the client - the deeper the streams API recursion goes.

In my particular application of the Streams API, I may too easily run into the recursion limits imposed by browsers.

How can I use the Streams API without recursion?  

Full details of my problem and an attempt at a solution (iterative use of the Streams API) are currently documented in this Stack Overflow question:

https://stackoverflow.com/questions/52771970/use-fetch-streams-api-to-consume-chunked-data-asynchronously-without-using-recur

I'd greatly appreciate any pointers you can give me - I'm new to this, so it may be staring me in the face?

QA

**Comment by ricea** [comment]
Recursion isn't needed. I answered in detail at https://stackoverflow.com/a/52810397/2523224.

**Comment by qacollective** [comment]
Thanks for a thorough answer. In the end, the culprit was actually the Chrome/Firefox debuggers themselves!! I changed my Console.Log to document.write for debug output, closed the debugger and the first code sample worked fine with both large and small chunks. My code did already handle split characters/JSON, although not elegantly. So I did a re-write as suggested: pastebin.com/hWGtQZvA but it wasn't as performant as my original code (surprisingly). So the fastest and 100% native code ended up as pastebin.com/Dr1CQVa2 which I'll now further refine. Thanks!
